src/main.rs

```diff
@@ -142,7 +158,17 @@ impl State {
         PrivateKeyInfo::from_der(key.as_ref())?;
         Certificate::from_der(crt.as_ref())?;
 
-        Ok(State { crt, san, key })
+        let config_obj = match config {
+            Some(c) => toml::from_str(&std::fs::read_to_string(Path::new(&c))?)?,
```

Comment:
- Both of these fallible methods require `.context` with cause.
- The code flow is not clear because the logic is jumping around, either split this into several lines or use a `.map`
- `Path::new` should never be necessary with sound typing, in fact `c` should work just fine, because `String` [implements `AsRef<Path>`](https://doc.rust-lang.org/std/path/struct.Path.html#impl-AsRef%3CPath%3E-for-String), expected by [`fs::read_to_string`](https://doc.rust-lang.org/std/fs/fn.read_to_string.html)
- `match` is redundant here, use `if let Some(config) = config { ... } else { Config::default() }`, or better yet, find a way to avoid explicit `Config::default` and just use `unwrap_or_default` on the option https://doc.rust-lang.org/nightly/core/option/enum.Option.html#method.unwrap_or_default if you can

Reply:
Resolved during our call today.